ksql-parser/src/test/java/io/confluent/ksql/parser/rewrite/StatementRewriteForStructTest.java

```diff
@@ -0,0 +1,133 @@
+import io.confluent.ksql.parser.tree.SingleColumn;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.util.DataSourceExtractor;
+import io.confluent.ksql.util.MetaStoreFixture;
+import org.junit.Before;
+import org.junit.Test;
+
+public class StatementRewriteForStructTest {
+
+  private static final KsqlParser KSQL_PARSER = new KsqlParser();
+
+  private MetaStore metaStore;
+
+  @Before
+  public void init() {
+    metaStore = MetaStoreFixture.getNewMetaStore(new TestFunctionRegistry());
+  }
+
+  @Test
+  public void shouldCreateCorrectFunctionCallExpression() {
+    final String simpleQuery = "SELECT iteminfo.category.name, address.state FROM orders;";
+    final Statement statement = KSQL_PARSER.buildAst(simpleQuery, metaStore).get(0);
+
+    assertThat( statement, instanceOf(Query.class));
+    final Query query = (Query) statement;
+    assertThat( query.getQueryBody(), instanceOf(QuerySpecification.class));
+    final QuerySpecification querySpecification = (QuerySpecification)query.getQueryBody();
+    assertThat( querySpecification.getSelect().getSelectItems().size(), equalTo(2));
+    final Expression col0 = ((SingleColumn) querySpecification.getSelect().getSelectItems().get(0)).getExpression();
+    final Expression col1 = ((SingleColumn) querySpecification.getSelect().getSelectItems().get(1)).getExpression();
+
+    assertThat(col0, instanceOf(FunctionCall.class));
+    assertThat(col1, instanceOf(FunctionCall.class));
+
+    assertThat(col0.toString(), equalTo("FETCH_FIELD_FROM_STRUCT(FETCH_FIELD_FROM_STRUCT(ORDERS.ITEMINFO, 'CATEGORY'), 'NAME')"));
+    assertThat(col1.toString(), equalTo("FETCH_FIELD_FROM_STRUCT(ORDERS.ADDRESS, 'STATE')"));
+
+  }
+
+  @Test
+  public void shouldNotCreateFunctionCallIfNotNeeded() {
+    final String simpleQuery = "SELECT orderid FROM orders;";
+    final Statement statement = KSQL_PARSER.buildAst(simpleQuery, metaStore).get(0);
+
+    assertThat( statement, instanceOf(Query.class));
+    final Query query = (Query) statement;
+    assertThat( query.getQueryBody(), instanceOf(QuerySpecification.class));
+    final QuerySpecification querySpecification = (QuerySpecification)query.getQueryBody();
+    assertThat( querySpecification.getSelect().getSelectItems().size(), equalTo(1));
+    final Expression col0 = ((SingleColumn) querySpecification.getSelect().getSelectItems().get(0)).getExpression();
+
+    assertThat(col0, instanceOf(DereferenceExpression.class));
+    assertThat(col0.toString(), equalTo("ORDERS.ORDERID"));
+  }
+
+
+  @Test
+  public void shouldCreateCorrectFunctionCallExpressionWithSubscript() {
+    final String simpleQuery = "SELECT arraycol[0].name as n0, mapcol['key'].name as n1 FROM nested_stream;";
+    final Statement statement = KSQL_PARSER.buildAst(simpleQuery, metaStore).get(0);
+    final SingleStatementContext singleStatementContext = KSQL_PARSER.getStatements(simpleQuery).get(0);
+
+    assertThat( statement, instanceOf(Query.class));
```

Comment:
```
assertThat(statement, instanceOf(Query.class));
    final Query query = (Query) statement;
    assertThat(query.getQueryBody(), instanceOf(QuerySpecification.class));
    final QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
```

Is repeated through each test.  Suggest moving to a method in the class that returns the querySpecification.

Reply:
Moved it to a method.